provider/digitalocean/digital_ocean.go

```diff
@@ -283,16 +290,26 @@ func makeDomainEditRequest(domain, name, recordType, data string, ttl int) *godo
 
 	// For some reason the DO API requires the '.' at the end of "data" in case of CNAME request.
 	// Example: {"type":"CNAME","name":"hello","data":"www.example.com."}
-	if recordType == endpoint.RecordTypeCNAME && !strings.HasSuffix(data, ".") {
+	if (recordType == endpoint.RecordTypeCNAME || recordType == endpoint.RecordTypeMX) && !strings.HasSuffix(data, ".") {
 		data += "."
 	}
 
-	return &godo.DomainRecordEditRequest{
+	request := &godo.DomainRecordEditRequest{
 		Name: adjustedName,
 		Type: recordType,
 		Data: data,
 		TTL:  ttl,
 	}
+
+	if recordType == endpoint.RecordTypeMX {
+		priority, domain, err := parseMxTarget(data)
+		if err == nil {
+			request.Priority = int(priority)
+			request.Data = provider.EnsureTrailingDot(domain)
+		}
```

Comment:
When `error != nil`, it should be (at least) displayed to the user and/or sent to the calling method.

Reply:
Basically after will be a raise from DO client. But I can add a logging.